sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala

```diff
@@ -89,5 +90,10 @@ case class Sum(child: Expression) extends DeclarativeAggregate with ImplicitCast
     )
   }
 
-  override lazy val evaluateExpression: Expression = sum
+  override lazy val evaluateExpression: Expression = dataType match {
+    case DecimalType.Fixed(precision, scale) =>
+      CheckOverflow(sum, DecimalType(precision, scale), SQLConf.get.decimalOperationsNullOnOverflow)
```

Comment:
`dataType` actually calls `resultType` but the latter reads better for our purpose here. Will update so.

Reply:
yes, sorry, I meant something like this:
```
    case _: DecimalType =>
      CheckOverflow(sum, resultType, SQLConf.get.decimalOperationsNullOnOverflow)
```
In this way we do not create a new `DecimalType` and I think it is more readable... Thanks.

Reply:
Updated to using matched case because `resultType` is still a generic DataType and not Decimal type
```
  override lazy val evaluateExpression: Expression = resultType match {
    case d: DecimalType => CheckOverflow(sum, d, SQLConf.get.decimalOperationsNullOnOverflow)
    case _ => sum
  }
```